### PR TITLE
Swiping tabs: Feature flag

### PR DESCRIPTION
**Pull request by 0nko**
Task/Issue URL: https://app.asana.com/0/1202552961248957/1208659739219426/f

### Description

This PR adds a feature flag that will allow us to disable the tab swiping.

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/browser/SwipingTabsFeature.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.feature.toggles.api.Toggle
+
+@ContributesRemoteFeature(
+    scope = AppScope::class,
+    featureName = "swipingTabs",
+)
+interface SwipingTabsFeature {
+    @Toggle.DefaultValue(false)
+    fun self(): Toggle
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/browser/SwipingTabsFeature.kt:29]: Don't forget to enable this flag along with a `minSupportedVersion` once the feature is released.

File: app/src/main/java/com/duckduckgo/app/browser/RealTabSwipingAvailability.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+@ContributesBinding(AppScope::class)
+class RealTabSwipingAvailability @Inject constructor(
+    swipingTabsFeature: SwipingTabsFeature,
+    private val dismissedCtaDao: DismissedCtaDao,
+) : TabSwipingAvailability {
+    private val isOnboardingComplete: Boolean
+        get() = dismissedCtaDao.exists(CtaId.DAX_END)
```
Comment (nalcalag) [inline app/src/main/java/com/duckduckgo/app/browser/RealTabSwipingAvailability.kt:31]: @0nko for checking if onboarding is finished, you need to check `UserStageStore.daxOnboardingActive()` instead `dismissedCtaDao.exists(CtaId.DAX_END)`. There are some scenarios where DAX_END dialog has been shown but onboarding is still active.
Also, this method is not used so not sure if it should be part of the PR?

File: app/src/main/java/com/duckduckgo/app/browser/RealTabSwipingAvailability.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    override val isSwipingTabsEnabled: Boolean = swipingTabsFeature.self().isEnabled() && isOnboardingComplete
+}
```
Comment (nalcalag) [inline app/src/main/java/com/duckduckgo/app/browser/RealTabSwipingAvailability.kt:34]:
```suggestion
@ContributesBinding(AppScope::class)
class RealTabSwipingAvailability @Inject constructor(
    swipingTabsFeature: SwipingTabsFeature,
    private val userStageStore: UserStageStore,
) : TabSwipingAvailability {
    override val isSwipingTabsEnabled: Boolean = swipingTabsFeature.self().isEnabled() && userStageStore.daxOnboardingActive()
}
```

File: app/src/main/java/com/duckduckgo/app/browser/RealTabSwipingAvailability.kt (hunk 2 above)
Comment (0nko) [reply app/src/main/java/com/duckduckgo/app/browser/RealTabSwipingAvailability.kt:31]: Thanks!

**Comment by 0nko** [comment]
Closing in favor of #5263

**Pull request closed without merging**